.github/workflows/ci.yml

```diff
@@ -58,7 +58,7 @@ jobs:
                       **/requirements*.txt
 
             - name: Install dependencies for test
-              run: pip install -U -e .[dev]
+              run: pip install -r requirements_test.txt
```

Comment:
I don't understand the need for such a change. All `dev` requirements have to be installed here anyways. Why create a separate file for that?

Reply:
The issue is that pip cannot install *only* the dev requirements, it installs all the main requirements *and* Dev. But the tests themselves run in the container so we don't need to install AWB here. This change saves 1 minute of CI time per job, which is significant given the overall time is around 4 minutes.

I have no issue of having `dev` together with `test`. Although that would also mean we install test requirements in the pre-commit CI job.

Reply:
I think we **do** want to test the installation. What would it be for a test otherwise?

I would rather work on python dependencies. As you've pointed out `pandas` takes quite a lot of time.

Reply:
> @unkcpz I assume this was your motivation in QeApp as well?

This is one aspect for QeApp. It has another problem if install with `pip install .[test]` in QeApp that it will check the dependencies and the QeAppWorkChain may lead to the conflicting. 

> I think we do want to test the installation. What would it be for a test otherwise?

The test is done inside the container by `aiidalab_exec` fixture.

Reply:
> The test is done inside the container by `aiidalab_exec` fixture.

This is done for the basic testing of the main features of AWB (I guess it is classified as _smoke testing_).

However, we plan to add unit tests to check individual components of the package that @danielhollas was mentioning during the AiiDAlab meeting. That won't run inside the container, right? The individual functions will be executed with a specific input and checked against their output. The package dependencies are still needed.

Reply:
The other option would be to run **everything** inside the container. I think this is doable as well. There is no need to run pytest outside of the docker container. I guess that everything (chrome, firefox, pytest, awb, notebook) could be kept inside of the container and checked from there.